packages/web-service/src/pages/habitat/grid-ref/habitat-grid-ref.js

```diff
@@ -1,6 +1,38 @@
+import Joi from 'joi'
 import pageRoute from '../../../routes/page-route.js'
 import { habitatURIs } from '../../../uris.js'
 
 export const completion = async _request => habitatURIs.WORK_START.uri
 
-export default pageRoute({ page: habitatURIs.GRID_REF.page, uri: habitatURIs.GRID_REF.uri, completion })
+export const validator = async payload => {
+  const habitatGridKey = 'habitat-grid-ref'
+  const gridRef = payload[habitatGridKey]
+
+  if (gridRef === '') {
+    throw new Joi.ValidationError('ValidationError', [{
+      message: 'Error: no grid ref has been sent',
+      path: [habitatGridKey],
+      type: 'no-reference-sent',
+      context: {
+        label: habitatGridKey,
+        value: 'Error',
+        key: habitatGridKey
+      }
+    }], null)
+  } else if (/[a-zA-Z]{2}\d{6}/.test(gridRef) === false) {
```

Comment:
Not sure why you are doing this. Just use the Joi.object in the validator arg as before

Reply:
Changed it